### PR TITLE
Documentation using mk-docs

### LINKED ISSUE #50: Design documentation site
The plugin will have a site for documentation, that will be updated per plugin release.
The documentation site will be created using the [GitHub Pages ](https://pages.github.com/) feature and the [ mkdocs tools ](https://squidfunk.github.io/mkdocs-material/) in order to automate the content generation from plugin release and deployments.

The site theme is to be decided by the developer choosen design, it should also cover the plugin project background, development section, plugin API and Usage, tutorial sections and plugin features.

### PR DESCRIPTION
**Pull request by vermeulendivan**
Fixes #50 

Documenation using mk-docs. This is only a starting point as there are too little to work with.

- Mk-docs has been created
- A basic structure and template has been used for now
- A quick introduction based on the user-stories document, which will like change in the future
- Added workflow and model figures

Plugin development has not started yet, so not much more to add until there are more information, etc. The theme, figures, paragraphs can easily be changed as required in the future.

image

image

### CODE REVIEW

**Comment by vermeulendivan** [comment]
Site look has been changes, nav bar reordered. Tabs has been added, for future use.

Link to the repo has been added.

image

File: .idea/.gitignore
Comment (Samweli) [inline .idea/.gitignore:1]: This should be removed by gitignore

File: docs/img/ci_logo.png
Comment (Samweli) [inline docs/img/ci_logo.png:1]: Do we have an svg for the icon?

File: docs/about/about.md
Comment (Samweli) [inline docs/about/about.md:1]: We should fill this section with content and other sections that we already have information for them.

**Review by Samweli — CHANGES_REQUESTED** [review]

File: .idea/.gitignore
Comment (vermeulendivan) [reply .idea/.gitignore:1]: @Samweli PR which adds the gitignore file, as there are none. Its and exact duplicate of the gitignore template for Python, and the .idea (for pycharm) folder has been added. https://github.com/kartoza/cplus-plugin/pull/124

Comment (vermeulendivan) [reply .idea/.gitignore:1]: idea folder removed

File: docs/img/ci_logo.png
Comment (vermeulendivan) [reply docs/img/ci_logo.png:1]: All png files has been replaced by svg files

File: docs/about/about.md
Comment (vermeulendivan) [reply docs/about/about.md:1]: Content has been added to the about section which provides a description - from the project pdf, with minor changes.

They have a nice About section on their website, we can maybe use this. But first have to ask permission. https://www.conservation.org/about

**Comment by vermeulendivan** [comment]
This PR has been changed to a draft.

File: .github/workflow/ci.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: ci
+on:
+  push:
+    branches:
+      - master
```
Comment (Samweli) [inline .github/workflow/ci.yml:5]: why adding a master branch?

File: .github/workflow/ci.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: ci
```
Comment (Samweli) [inline .github/workflow/ci.yml:1]: This file should only be used for testing purposes, we should use the build-docs.yml for documentation configurations.

File: .gitignore [hunk 3]
```diff
@@ -129,4 +129,3 @@ dmypy.json
 
 # Development
 .idea/
-build/
```
Comment (Samweli) [inline .gitignore:132]: The build folder is meant to be ignore by git

Comment (vermeulendivan) [reply .gitignore]: there is 2 build/ in the gitignore file

File: .github/workflow/ci.yml (hunk 1 above)
Comment (vermeulendivan) [reply .github/workflow/ci.yml:5]: This is how its set up from the explanations on mkdocs, likely copied it from my test example and did not realize it. Will remove it.

File: .github/workflow/ci.yml (hunk 2 above)
Comment (vermeulendivan) [reply .github/workflow/ci.yml:1]: This is how its set up from the mkdocs site, stayed with their structure. Will have a look if it causes any issues if I change it.

Comment (vermeulendivan) [reply .github/workflow/ci.yml]: Done

File: .gitignore (hunk 3 above)
Comment (vermeulendivan) [reply .gitignore]: Ignore file the same as the main ignore file

File: .github/workflow/ci.yml (hunk 1 above)
Comment (vermeulendivan) [reply .github/workflow/ci.yml]: Done

File: .github/workflow/mk-docs.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+name: mk-docs
```
Comment (Samweli) [inline .github/workflow/mk-docs.yml:1]: is any difference between this file and the `build-docs.yml`?

Comment (vermeulendivan) [reply .github/workflow/mk-docs.yml:1]: If I look at it, yes. This is put together from how its shown on mkdocs. Do you want me to move this into the build-docs.yml?

Comment (Samweli) [reply .github/workflow/mk-docs.yml:1]: Yes lets move it to `build-docs.yml` and use `requirements-dev` for installation

Comment (vermeulendivan) [reply .github/workflow/mk-docs.yml:1]: Will do

Comment (vermeulendivan) [reply .github/workflow/mk-docs.yml]: Made the change

File: .github/workflows/build-docs.yml [hunk 5]
```diff
@@ -1,21 +1,39 @@
...
+          key: ${{github.ref}}
+          path: .cache
 
       - name: Fix Python command
         run: apt-get install python-is-python3
 
       - name: Install plugin dependencies
         run: pip install -r requirements-dev.txt
 
+      - name: Install mkdocs-material
```
Comment (Samweli) [inline .github/workflows/build-docs.yml:34]: We don't need this step as it is already available in the requirements-dev file

Comment (vermeulendivan) [reply .github/workflows/build-docs.yml]: Done

**Review by Samweli — APPROVED** [review]

**Pull request merged**